core/commands/diff.py

```diff
@@ -223,7 +300,10 @@ def run(self, edit, setting):
 
         setting_str = "git_savvy.diff_view.{}".format(setting)
         current_mode = settings.get(setting_str)
-        next_mode = not current_mode
+        if setting == 'show_word_diff':
+            next_mode = (current_mode + 1) % len(WORD_DIFF_PATTERNS)
+        else:
+            next_mode = not current_mode
```

Comment:
I like this, but maybe it should be a separate command, e.g `GsDiffCycleSetting` ?